app/push-notification/page.tsx

```diff
@@ -1,34 +1,91 @@
-import React from "react";
+"use client";
+import React, { useEffect, useState } from "react";
 import DashboardLayout from "../layouts/Dashboard";
 import PushInfomation from "@/components/PushInfomation";
 import { Button } from "@/components/ui/button";
 import Editor from "@/components/Editor";
-import { DatePicker } from "@/components/DatePicker";
-import QuillToolbar from "@/components/Toolbar";
-import { notificationData } from "@/types/data";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { toast } from "sonner";
+import { getNotifications, pushNotification } from "@/API/notiifications";
+import { dateFormat } from "@/lib/dateFormat";
+
+interface Notification {
+  createdAt: string;
+  title: string;
+  message: string;
+  _id: string;
+}
 
 export default function NotificationPage() {
+  const queryClient = useQueryClient();
+  const [value, setValue] = useState("");
+  const [title, setTitle] = useState("");
+  const [sendToAll, setSendToAll] = useState(false);
+  const [date, setDate] = useState<Date | undefined>();
```

Comment:
you using parent component state to handle the date from child component this might trigger get query as well check it out this is not the right way to do it divide the push notification  into 2 parts fetch data in the other parts where you have to show only previouse notification